internal/libhive/testmanager.go

```diff
@@ -169,55 +168,84 @@ func (manager *TestManager) GetNodeInfo(testSuite TestSuiteID, test TestID, node
 
 // CreateNetwork creates a docker network with the given network name, returning
 // the network ID upon success.
-func (manager *TestManager) CreateNetwork(testSuite TestSuiteID, networkName string) (string, error) {
+func (manager *TestManager) CreateNetwork(testSuite TestSuiteID, name string) error {
 	_, ok := manager.IsTestSuiteRunning(testSuite)
 	if !ok {
-		return "", ErrNoSuchTestSuite
+		return ErrNoSuchTestSuite
 	}
 
 	// add network to network map
 	manager.networkMutex.Lock()
 	defer manager.networkMutex.Unlock()
 
-	netID, err := manager.backend.CreateNetwork(networkName)
+	unique := getUniqueName(testSuite, name)
+
+	id, err := manager.backend.CreateNetwork(unique)
 	if err != nil {
-		manager.networks[netID] = networkName
+		return err
 	}
-	return netID, err
+	if _, exists := manager.networks[testSuite]; !exists {
+		// initialize network map for individual test suite
+		manager.networks[testSuite] = make(map[string]string)
+		// add bridge network information
+		bridgeID, err := manager.backend.NetworkNameToID("bridge")
```

Comment:
I'm not sure about this. I think it would be easier to resolve "bridge" the same way as before, i.e. just check for the special name in ContainerIP.